src/imagery/i.eodag/i.eodag.py

```diff
@@ -383,54 +390,58 @@ def get_aoi(vector=None):
                 for poly_coords in coord_proj.outputs["stdout"]
                 .value.strip()
                 .split("\n")
-            ]
+            ],
         ) + "))"
         return projected_geom
     return geom
 
 
-def search_by_ids(products_ids):
+def search_by_ids(ids_set: set, module_options: dict, eodag_api=None):
     """Search for products based on their ids.
 
-    :param products_ids: List of products' ids.
-    :type products_ids: list
+    :param ids_set: Set of unique products' ids.
+    :type ids_set: list
+    :param module_options: Dict with GRASS module options.
+    :type module_options: dict
 
     :return: EO products found by searching with 'search_parameters'
     :rtype: class:'eodag.api.search_result.SearchResult'
     """
-    gs.verbose("Searching for products...")
+    # Remove empty string
+    ids_set.discard("")
+    # Search for products found from options["file"] or options["id"]
+    gs.verbose(_("Searching for {} distinct ID(s).").format(len(ids_set)))
+    # Search for products found from options["file"] or options["id"]
     search_result = []
-    for query_id in products_ids:
+    for query_id in ids_set:
         gs.info(_("Searching for {}").format(query_id))
-        if int(eodag.__version__.split(".")[0]) < 3:
-            product, count = dag.search(
-                id=query_id, provider=options["provider"] or None
-            )
-        else:
-            if options["producttype"] is None:
-                gs.warning(_("The producttype option is not set"))
-            product = dag.search(
-                id=query_id,
-                provider=options["provider"] or None,
-                productType=options["producttype"] or None,
-                count=True,
-            )
-            count = product.number_matched
-        if count > 1:
+        if module_options["producttype"] is None:
+            gs.warning(_("The producttype option is not set"))
+        product = eodag_api.search(
+            id=query_id,
```

Comment:
```suggestion
        if not module_options["producttype"]:
            gs.warning(_("The producttype option is not set"))
        product = eodag_api.search(
            id=query_id,
```

Would be an empty string.